sdk/storage/Azure.Storage.Common/src/Shared/StorageProgressExtensions.cs

```diff
@@ -9,9 +9,24 @@ namespace Azure.Storage
     internal static class StorageProgressExtensions
     {
         public static Stream WithProgress(this Stream stream, IProgress<long> progressHandler)
-            =>
-                progressHandler != null && stream != null
-                    ? new AggregatingProgressIncrementer(progressHandler).CreateProgressIncrementingStream(stream)
-                    : stream;
+        {
+            if (progressHandler != null && stream != null)
+            {
+                if (progressHandler is AggregatingProgressIncrementer)
```

Comment:
nit - can avoid doing two type casts:
```
     if (progressHandler is AggregatingProgressIncrementer handler)  
          {  
               return handler.CreateProgressIncrementingStream(stream);
          }  
```

Reply:
The 2nd cast is necessary, or we get a compile error.

Reply:
Did you also change the second line to use `handler` instead of `progressHandler`?  What compiler error were you getting because that should work.

Reply:
Fixed.